tests/Integration/ApiPlatform/DiscountEndpointTest.php

```diff
@@ -434,4 +434,24 @@ public function testGetUpdatedDiscountConditions(int $discountId): void
 
         $this->addToAssertionCount(1);
     }
+
+    /**
+     * @depends testCreateDiscountForUpdateTests
+     *
+     * @param int $discountId
+     *
+     * @return void
+     */
+    public function testDuplicateDiscount(int $discountId): void
+    {
+        $newDiscount = $this->createItem(
+            '/discounts/' . $discountId . '/duplicate',
+            [],
+            ['discount_write'],
+            201
+        );
+
+        $this->assertArrayHasKey('discountId', $newDiscount);
+        $this->assertNotEquals($discountId, $newDiscount['discountId']);
```

Comment:
Maybe we can improve the test to add some other checks (to check the copied properties)